### ISSUE TITLE
The new dataset 

### ISSUE THREAD

**Issue by Daybreak-Zheng** [SadilKhan/Text2CAD#11]
Great job! I have a new question about dataset.
Why are the **Level 2** data in "**all_level_data**: (All the level annotations generated by LLM before splitting)" and the data in the separately listed "**intermediate: Intermediate Level Annotation (Level 2)**" not the same? 
What's the difference between them?
**1. What's the meaning of "before splitting" in "all_level_data**: (All the level annotations generated by LLM before splitting)"?**

<img width="1051" alt="Clipboard_Screenshot_1736240822" src="https://github.com/user-attachments/assets/9ee35c5e-b31d-4804-97e9-c987bba95017" />

Besides, I've found that in the dataset of version v1.1, the train split has been reduced by a large amount. 

<img width="780" alt="Clipboard_Screenshot_1736240232" src="https://github.com/user-attachments/assets/227daacc-bbbb-4d23-a061-a241bf4c3708" />

**2. Will reducing these cuboid and cylinders lead to better results for the training model?**

**3. Will the new dataset v1.1 have better results? When will you probably update the experimental results of v1.1?**
I am really looking forward to it.

**Comment by Daybreak-Zheng** [comment]
Thank you for your reply. 
When training the v1.1 dataset, can I simply clone the code from your GitHub repo and use the default config? 
Are there any parameters I need to modify?
I am really looking forward to it.

**Comment by SadilKhan** [comment]
Yes. Default parameter is fine. Just adapt the batch size to your gpu memory(i kept it low for testing). Also, in trainer.yaml, provide your own directory and paths.

**Comment by Daybreak-Zheng** [comment]
What batch size did you use during your training with A100? 
When I use a smaller batch size with the RTX 3090, do I need to adjust the learning rate? Will this affect the final training results?

**Comment by SadilKhan** [comment]
You can try with 96 or 128. I used batch size around 320 in A100 80GB once.

**Comment by Daybreak-Zheng** [comment]
I found that when I set the batch size to 160 and the gradient accumulation to 320 on the A100 40GB on v1.1 dataset, the training is very slow. Is this normal?
<img width="814" alt="Clipboard_Screenshot_1736770347" src="https://github.com/user-attachments/assets/8dfa9122-4d9f-44d0-a3b5-326f18074c4b" />
And it takes three hours just to load the data.
<img width="567" alt="Clipboard_Screenshot_1736770546" src="https://github.com/user-attachments/assets/d565d655-cb2f-43a3-ba7d-ac537c831f85" />

I see that you mentioned in your paper that **training for 160 epochs only takes 2 days**.
<img width="705" alt="Clipboard_Screenshot_1736770591" src="https://github.com/user-attachments/assets/9e16f741-8fcf-44ed-b714-4a2e78cf8d24" />
 I wonder if I might have set something incorrectly.

**Comment by SadilKhan** [comment]
Hello,
During the submission, we did something different to train faster.  We saved the text embedding from bert beforehand and also used litdata to shard the dataset. I will update it in the camera ready version of our paper. However, there is a storage complexity if you use this, hence I removed them from the final code. 

There are couple of things you can try. 
1. Increase the number of workers( use 16/24/32). 
2. Save the `self.prompt_data` [here](https://github.com/SadilKhan/Text2CAD/blob/main/Cad_VLM/dataprep/t2c_dataset.py#L66) locally. Then you need to load the data only once. Next time, just use the local data.
4. You can shard the data using litdata. I will add it in future as an optional in the code. 
5. The current code takes 1h30mins in L40S and 30 mins in H100.

Can you also please share whether your gpu utilisation goes to 0 sometimes and if yes, does it stay like this for a long time?

**Comment by SadilKhan** [comment]
Yes, the workers refer to the num work of dataloader. 

Litdata is used for sharding the dataset. You can check [here](https://github.com/Lightning-AI/litdata?tab=readme-ov-file#features-for-optimizing-and-streaming-datasets-for-model-training) - **Encrypt, decrypt data at chunk/sample level**. After Sharding, you can use litdata’s own dataloader to load the dataset. 
I really appreciate your interest in Text2CAD but as I said, I am prioritising other projects now due to an upcoming rebuttal. Unfortunately, I still haven’t got the time yet to do this but I will soon add it.
